docs/guides/src/main/server/health.adoc

```diff
@@ -34,6 +34,35 @@ It is possible to enable the health checks using the build time option `health-e
 
 By default, no check is returned from the health endpoints.
 
+== Using the health checks
+
+It is recommended that the health endpoints be monitored by external HTTP requests. Due to security measures that remove `curl` and other packages from the Keycloak container image, local command-based monitoring will not function easily.
+
+If you are not using Keycloak in a container, use whatever you want to access the health check endpoints.
+
+=== curl
+
+You may use a simple HTTP HEAD request to determine the `+live+` or `+ready+` state of Keycloak. `+curl+` is a good HTTP client for this purpose.
+
+If Keycloak is deployed in a container, you must run this command from outside it due to the previously mentioned security measures. For example:
+
+[source, bash]
+----
+curl --head -fsS http://localhost:8080/health/ready
+----
+
+If the command returns with status 0, then Keycloak is `+live+` or `+ready+`, depending on which endpoint you called. Otherwise there is a problem.
+
+=== Kubernetes
+
+Define a https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#http-probes[HTTP Probe] so that Kubernetes may externally monitor the health endpoints. Do not use a liveness command.
+
+=== HEALTHCHECK
+
+The Dockerfile image `+HEALTHCHECK+` instruction defines a command that will be periodically executed inside the container as it runs. The Keycloak container does not have any CLI HTTP clients installed. An executable jar, using the built-in Java HTTP client is nessessary to use the health endpoint in this way.
+
+No such jar is provided as part of the Keycloak installation. You will have to implement it yourself, or use an alternative method.
```

Comment:
I'm still not really sure we should instruct users to write their own `curl` in Java, even performance wide it would not be ideal. Just a stupid question but can't we simply [download `curl`](https://curl.se/download.html) to avoid the RPM dependencies mess?

Reply:
The linked zip is just the source code. Compiling our own curl with just HTTP support would solve the problem, we could even use nss for curl's HTTPS support (the RPM version uses openssl, doubling the SSL library surface), but compiling our own is also not ideal in of itself.

It is possible to do HTTP with just bash, I almost included the example here, but the problem is HTTPS. TLS support would require the openssl command. I can't find an equivalent CLI TLS interface for nss.

```bash
exec 3<>/dev/tcp/example.com/80
echo -e "GET / HTTP/1.1\r\nHost: example.com\r\nConnection: close\r\n\r\n" >&3
cat <&3
```

If the server could be configured to maintain HTTP bound to loopback in production mode, that'd help. Everything else would still be forced to use HTTPS

Reply:
If I'm not mistaken, there are binaries on the download link but e.g. for Fedora/RHEL it's still distributed as RPM, so doesn't solve anything.

Then I'd probably instruct users to consider installing the `curl` from dnf (as an alternative to Java, which is IMHO too heavy for a simple health check).